### ISSUE TITLE
Avoid returning `null`.  Prefer `Optional`

### ISSUE THREAD

**Issue by randykerber** [bridgedb/BridgeDb#63]
In situations where a value might not exist, generally preferable to return values wrapped with a Java `Optional<>` rather than a `null`.  It's more self documenting, and generally easier to deal with logic to handle situations where values might be missing.

**Comment by stain** [comment]
Yes, no deliberate `null` returns. Which places did you identify this?

`null` might still make sense as *input* parameter though.
